lib/miq_automation_engine/engine/miq_ae_workspace.rb

```diff
@@ -60,12 +61,39 @@ def readonly?
       @readonly
     end
 
+    def self.current=(ws)
+      Thread.current.thread_variable_set(:current_workspace, ws)
```

Comment:
Is `thread_variable_set` needed or could you use `Thread.current[: current_workspace] = ws`.  

Applies to the matching `thread_variable_get` below as well.